x-pack/plugins/observability_solution/apm/public/components/alerting/ui_components/alert_details_app_section/latency_chart.tsx

```diff
@@ -247,7 +240,7 @@ function LatencyChart({
         </EuiFlexGroup>
         <TimeseriesChart
           id="latencyChart"
-          annotations={getLatencyChartAdditionalData()}
+          annotations={[...getLatencyChartAdditionalData(), ...getLatencyChartAlertStartData()]}
```

Comment:
should this annotation be wrapped in a `useMemo`? I'm not sure if building a new array here could cause unnecessary re-renders.

Reply:
This annotation was already being passed a new reference each time via the `getLatencyChartAdditionalData`, so this implementation is quite similar.

I don't think this meets the criteria for valuable use of `useMemo` based upon this reading https://react.dev/reference/react/useMemo#should-you-add-usememo-everywhere. The calculation here isn't particularly expensive. Also, this component is only is pretty static and doesn't experience many rerenders because the alert details page itself is pretty static (no filters, time range, or search). So I don't think it's necessary to add `useMemo` here.